op-batcher/rpc/api.go

```diff
@@ -34,6 +38,36 @@ func GetAdminAPI(api *adminAPI) gethrpc.API {
 	}
 }
 
+func (a *adminAPI) UseDAType(
+	ctx context.Context,
+	kind flags.DataAvailabilityType,
+	frameCount *uint,
+) error {
+	if !flags.ValidDataAvailabilityType(kind) {
+		return fmt.Errorf("unknown data-availability type: %q", kind)
+	}
+	if kind == flags.BlobsType && frameCount == nil {
+		return fmt.Errorf("cant use blobs and have nil framecount second argument")
+	}
+	if kind == flags.CalldataType && frameCount != nil {
+		return fmt.Errorf("cant use calldata with framecount second argument")
+	}
+	if kind == flags.BlobsType && frameCount != nil && (*frameCount == 0 || *frameCount > 6) {
+		return fmt.Errorf("cant use blobs with frame count outside range [1,6] given: %d", *frameCount)
+	}
+
+	if err := a.b.SanityCheckConfigUpdate(kind, frameCount); err != nil {
+		return err
+	}
+
+	if err := a.b.StopBatchSubmitting(ctx); err != nil {
+		return err
+	}
+	// cant fail, we already sanity checked it
+	a.b.ChangeDataAvailability(kind, frameCount)
```

Comment:
Since `SanityCheckConfigUpdate` and `ChangeDataAvailability` are not coupled, it results in a future dev being able to call `ChangeDataAvailability` without calling sanity check. I think they should be coupled more tightly, ie sanity check is internal or the impl is directly inside change and change returns an error

Reply:
agree & initially I had `ChangeDataAvailability` return error if bad config but it would mean a potentially needless stopping of batch submitting.